### PR TITLE
Week3step5

### PR DESCRIPTION
**Pull request by jackdavidweber**

### CODE REVIEW

**Comment by g-harel** [comment]
This seems like it includes changes from the preceding PR?

**Comment by augerm** [comment]
> This seems like it includes changes from the preceding PR?

+1 if the other PR isn't ready to be merged yet, you can set the base branch to that one in the meanwhile for the PR.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,31 +23,91 @@
...
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
+    // only show number of comments specified
+    String json;
+    if (comments.size() < maxComments){
+        json=new Gson().toJson(comments);
+    } else {
+        json=new Gson().toJson(comments.subList(0,maxComments));
```
Comment (augerm) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: `json = new Gson()` for spacing consistency

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,31 +23,91 @@
...
+    }
 
-    response.setContentType("text/html;");
+    response.setContentType("application/json;");
```
Comment (augerm) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: Good catch!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -23,31 +23,91 @@
...
+    try {
+      commentsChoice = Integer.parseInt(commentsChoiceString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + commentsChoiceString);
+      return maxComments;
+    }
+
+    // Check that the input is greater than 1.
+    if (commentsChoice < 1) {
+      System.err.println("Player choice is out of range: " + commentsChoiceString);
```
Comment (augerm) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit/optional - `Player` --> `User`

**Review by augerm — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (jackdavidweber) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: got it!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (jackdavidweber) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: got it

**Pull request merged**